### PR TITLE
Add tests for deleting and editing addresses

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
Sorry, half of this is API mocks.

This adds RTL tests to make sure that happy and sad paths are handled for editing and deleting addresses.

## Testing done
Make sure the app properly handles editing and deleting addresses when the API behaves or misbehaves.

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx [hunk 1]
```diff
@@ -15,7 +15,6 @@ import {
   renderWithProfileReducers,
   wait,
 } from '../../unit-test-helpers';
-import { beforeEach } from 'mocha';
```
Comment (erikphansen) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx:18]: This was never needed; I think VSCode somehow added it

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-address.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    // the aria-label is more descriptive
+    editButton = view.queryByRole('button', {
+      name: new RegExp(`edit.*${addressName}`, 'i'),
+    });
+  }
+  return editButton;
+}
+
+function deleteAddress(addressName) {
+  getEditButton(addressName).click();
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-address.unit.spec.jsx:42]: Just a general thought, not related to this block per se - it feels weird to me that the file extension is `unit.spec.jsx` while we're doing integration tests.. 

I do really appreciate your thoroughness with the tests, there just seems a bit of a mismatch if you know what I mean 😄 I don't really know of an action item here though

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.update-address.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+  addresses.forEach(address => {
+    const addressName = FIELD_TITLES[address];
+    describe(`${addressName} when the entered address passes address validation`, () => {
+      it('should handle a transaction that succeeds quickly', async () => {
+        await testQuickSuccess(addressName);
+      });
+      it('should handle a transaction that does not succeed until after the edit view exits', async () => {
+        await testSlowSuccess(addressName);
+      });
+      // TODO: this test does not pass. If the address validation API fails, we exit edit mode when we should actually keep it open. So we should fix the edit address flow to make this pass.
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.update-address.unit.spec.jsx:260]: Good catch, do we have a ticket for this that also mentions to un-skip this test?

**Review by sanlouise — APPROVED** [review]
Great coverage! Looks good to me 👍

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-address.unit.spec.jsx (hunk 2 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-address.unit.spec.jsx:42]: Ha, yeah I think that's required by our `test:unit` script. Remember when we say "unit test" that just means "mocha test" 😁

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.update-address.unit.spec.jsx (hunk 3 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.update-address.unit.spec.jsx:260]: Good call. I will add that before I merge this

Comment (erikphansen) [reply src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.update-address.unit.spec.jsx]: I actually fixed this. I was setting up my API mock incorrectly :-|

**Pull request merged**